public/app/plugins/datasource/grafana-azure-monitor-datasource/utils/migrateQuery.ts

```diff
@@ -22,6 +22,7 @@ export default function migrateQuery(query: AzureMonitorQuery): AzureMonitorQuer
   workingQuery = migrateToDefaultNamespace(workingQuery);
   workingQuery = migrateMetricsDimensionFilters(workingQuery);
   workingQuery = migrateResourceUri(workingQuery);
+  workingQuery = migrateDimensionFilters(workingQuery);
```

Comment:
there is already a `migrateMetricsDimensionFilters` above. Can we reuse that method or is there any reason for having a different function here?

Reply:
I personally think the migrations should be kept independent of each other as it makes it easier to track what changes are being made to the query and when. Happy to discuss this and hear opinions on it 😊

Reply:
I agree, better to have different methods but here it would be helpful to know what's the difference between them. Maybe renaming would help here: the one in line 23 can be  `migrateDimensionToDimensionFilter` and this one `migrateDimensionFilterToArray`. WDYT?

Reply:
I think a rename is more appropriate in this case 😊